test/Erc20.ts

```diff
@@ -0,0 +1,107 @@
+import { time, loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import { anyValue } from "@nomicfoundation/hardhat-chai-matchers/withArgs";
+import { expect } from "chai";
+import { ethers } from "hardhat"
+import { SolicyCoinERC20 } from "../typechain-types";
+const BigNumber = require('bignumber.js');
+
+const Web3EthAccounts = require('web3-eth-accounts');
+
+const account = new Web3EthAccounts('ws://localhost:8546');
+
+
+describe("ERC20", function () {
+    
+    let erc20: SolicyCoinERC20
+
+    beforeEach(async () => {
+        const SolicyCoinERC20 = await ethers.getContractFactory("SolicyCoinERC20");
+        erc20 = await SolicyCoinERC20.deploy(5);
+    });
+
+    describe("Init", function () {
+        it ("The token Name is correct", async () => {
+            expect(await erc20.name()).to.equal("Solicy"); 
+        })
+        it ("The token Symbol is correct", async () => {
+            expect(await erc20.symbol()).to.equal("SOL"); 
+        })
+        it ("The token decimals set correct", async () => {
+            expect(await erc20.decimals()).to.equal(18); 
+        })
+    });
+
+    describe("Mint", () => {
+        it ("Essentially minted correct count of",async () => {
+            expect(await erc20.totalSupply()).to.equal(10 ** 10); 
+        })
+        it ("The Mint functional is blocked for everyone", async () => {
+            try {
+                await erc20.mint()
+            } catch (error: any) {
+                expect(error.errorArgs[0]).to.equal("Mint action blocked in this contract");
+            }
+        })
+    })
+
+    describe("Burn", function () {
+        it ("Token burn test complete successfully", async () => {
```

Comment:
You should have test case with error on burning  when user tries to burn other users token